retrieval/scrape.go

```diff
@@ -807,8 +879,11 @@ func (sl *scrapeLoop) reportStale(start time.Time) error {
 }
 
 func (sl *scrapeLoop) addReportSample(app storage.Appender, s string, t int64, v float64) error {
-	ref, ok := sl.refCache[s]
+	// Suffix s with the invalid \xff unicode rune to avoid collisions
+	// with scraped metrics.
+	s2 := s + "\xff"
```

Comment:
Are there any unclarities left here? This simply ensures the cache does not collide for unusual but valid cases. It does not prevent the user from doing completely funky stuff.